packages/connector-sendgrid-mail/src/types.ts

```diff
@@ -0,0 +1,119 @@
+import { Nullable } from '@silverhand/essentials';
+import { z } from 'zod';
+
+/**
+ * @doc https://docs.sendgrid.com/api-reference/mail-send/mail-send#body
+ */
+
+export enum ContextType {
+  Text = 'text/plain',
+  Html = 'text/html',
+}
+
+export type EmailData = {
+  name?: string;
+  email: string;
+};
+
+export type Personalization = {
+  to: EmailData[];
+  from?: EmailData;
+  cc?: EmailData | EmailData[];
+  bcc?: EmailData | EmailData[];
+  subject?: string;
+  headers?: Record<string, string>;
+  substitutions?: Record<string, string>;
+  dynamic_template_data?: Record<string, any>;
+  custom_args?: Record<string, string>;
+  sendAt?: number;
+};
+
+export type Content = {
+  type: 'text/html' | 'text/plain';
```

Comment:
We can convert `ContextType` enum to a union type, without having to define it again.
```typescript
type: `${ContextType}`
```